src/concurrent.rs

```diff
@@ -289,33 +300,58 @@ mod tests {
         assert_eq!(0, reader2.read(&mut buffer).expect("Expected 0 byte count as EOF after close"));
     }
 
-    #[test]
-    fn test_shareable_file_close_unblocks_reads_without_error() {
+    fn try_shareable_file_close_unblocks_reads_without_error() {
         let dir = tempfile::tempdir().unwrap();
         let path = dir.path().join("pipe");
         unistd::mkfifo(&path, sys::stat::Mode::S_IRUSR | sys::stat::Mode::S_IWUSR).unwrap();
 
-        // We need to write to the FIFO for the read-only open below to not block.
+        // We need to open the FIFO for writes so that the open for reads doesn't block.  But we
```

Comment:
Yeah, but then I don't think we would be testing what I need to test :)

We need the read to block because that's how the reconfiguration thread will behave, and the whole point of this test is verifying that we can unblock that read asynchronously. If I opened the file in non-blocking mode, that wouldn't be the case right?